src/libraries/Microsoft.PowerFx.Repl/Repl.cs

```diff
@@ -214,10 +219,13 @@ public async Task HandleLineAsync(string line, CancellationToken cancel = defaul
         /// </summary>
         /// <param name="expression">expression to run.</param>
         /// <param name="cancel">cancellation token.</param>
+        /// <param name="lineNumber">line number to attribute errors to.</param>
         /// <returns>status object with details.</returns>
         /// <exception cref="InvalidOperationException">invalid.</exception>
-        public virtual async Task<ReplResult> HandleCommandAsync(string expression, CancellationToken cancel = default)
+        public virtual async Task<ReplResult> HandleCommandAsync(string expression, CancellationToken cancel = default, uint? lineNumber = null)
```

Comment:
is lineNumber the right abstraction?
Is this just for printing source information?

Reply:
This is just for printing source information.  Completely optional, default null doesn't change any behavior.

I'm having the repl host pass this in, since it is responsible for pulling the lines out of a file and feeding them to repl one by one.  It also facilitates if the host does some preprocessing, say to pull islands of Power Fx out of a Jupyter like file.